### PR TITLE
Tests remove session

### LINKED ISSUE #848: IT Wazuh-logtest - Check remove session.
Hello team!

To test the Wazuh-logtest you can treat the behavior when sending messages through the `/var/ossec/queue/logtest` socket to remove sessions.

Best regards,
Juan Cabrera

### PR DESCRIPTION
**Pull request by JcabreraC**
Hello team,

This PR adds tests for remove session on demand in `Wazuh-logtest`. We implement tests that send remove session messages to logtest (writing the wazuh-logtest unix socket) and check error, warning or success message is expected.

Closes #848 

# Tests checks

- [x] Proven that tests **pass** when they have to pass
- [x] Proven that tests **fail** when they have to fail
- [x] Tested and passed in Jenkins.

# Results

- When the expected behavior occurs:
```
======================================= test session starts ========================================
platform linux -- Python 3.6.8, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.8', 'Platform': 'Linux-3.10.0-1062.4.3.el7.x86_64-x86_64-with-centos-7.7.1908-Core', 'Packages': {'pytest': '6.0.1', 'py': '1.9.0', 'pluggy': '0.13.1'}, 'Plugins': {'testinfra': '5.0.0', 'metadata': '1.8.0', 'html': '2.0.1'}}
rootdir: /vagrant/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: testinfra-5.0.0, metadata-1.8.0, html-2.0.1
collected 9 items                                                                                  

test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid numeric token] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid json] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid empty json object token] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid array token] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid token null] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid boolean token ] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove invalid token len] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove non-exist session] PASSED
test_logtest/test_remove_session/test_remove_session.py::test_remove_session[Remove session OK] PASSED

------------------------- generated html file: file:///vagrant/report.html -------------------------
======================================== 9 passed in 0.47s =========================================
```

Best regards,
Juan Cabrera

### CODE REVIEW

**Review by juliancnn — APPROVED** [review]

File: tests/integration/test_logtest/test_remove_session/data/remove_session.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+-
+  name: "Remove invalid token type"
```
Comment (jnasselle) [inline tests/integration/test_logtest/test_remove_session/data/remove_session.yaml:3]: I think "Remove invalid numeric token" define better the test case

File: tests/integration/test_logtest/test_remove_session/data/remove_session.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+-
+  name: "Remove invalid json"
+  description: "Check remove session"
+  test_case:
+  -
+    input: '{"remove_session": asd}'
+    output: "{\"messages\":[\"ERROR: (7306): Error parsing JSON\",\"ERROR: (7307): Error in position 19, ... session\\\": asd} ...\"],\"codemsg\":-2}"
+    stage: 'Remove invalid json'
+-
+  name: "Remove null token"
```
Comment (jnasselle) [inline tests/integration/test_logtest/test_remove_session/data/remove_session.yaml:19]: I think "Remove invalid empty json object token" define better the test case ( based on JSON Data types)

File: tests/integration/test_logtest/test_remove_session/data/remove_session.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+-
+  name: "Remove invalid token null"
+  description: "Check remove session"
+  test_case:
+  -
+    input: '{"remove_session":null}'
+    output: "{\"messages\":[\"ERROR: (7316): Failure to remove session. remove_session JSON field must be a string\"],\"codemsg\":-2}"
+    stage: 'Remove invalid token'
+-
+  name: "Remove invalid token"
```
Comment (jnasselle) [inline tests/integration/test_logtest/test_remove_session/data/remove_session.yaml:43]: I think "Remove invalid json object token" define better the test case ( based on JSON Data types)

File: tests/integration/test_logtest/test_remove_session/data/remove_session.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+-
+  name: "Remove null token"
+  description: "Check remove session"
+  test_case:
+  -
+    input: '{"remove_session": {}}'
+    output: "{\"messages\":[\"ERROR: (7316): Failure to remove session. remove_session JSON field must be a string\"],\"codemsg\":-2}"
+    stage: 'Remove null token'
+-
+  name: "Remove invalid token list"
```
Comment (jnasselle) [inline tests/integration/test_logtest/test_remove_session/data/remove_session.yaml:27]: I think "Remove invalid array token" define better the test case ( based on JSON Data types)

**Review by jnasselle — CHANGES_REQUESTED** [review]
I forgot to mention that also adding "Remove invalid boolean token" will gave us 100% coverage about token data types

**Review by jnasselle — APPROVED** [review]

**Pull request merged**